main.cpp

```diff
@@ -277,36 +166,89 @@ struct HeapA
 
  Wait for my code review.
  */
+
 #include <iostream>
 #include <string>
+#include <cmath>
 
 struct IntType;
 struct FloatType;
 struct DoubleType;
 
 struct FloatType
 {
-    FloatType(float f) : value(new float(f)) { }
+    explicit FloatType(float f) : value(new float(f)) { }
+
     ~FloatType()
     {
         delete value;
         value = nullptr;
     }
-    operator float()
-    {
-        return *value;
-    }
 
     FloatType& add( float rhs );
-   
     FloatType& subtract( float rhs );
-   
     FloatType& multiply( float rhs );
-   
     FloatType& divide( float rhs);
-   
+    operator float() const { return *value; }
+    FloatType& pow(const FloatType&);
+    FloatType& pow(const DoubleType&);
+    FloatType& pow(const IntType&);
+    FloatType& pow(float rhs);
+
     private:
         float* value = nullptr;
+        FloatType& powInternal(const float exp);
+};
+
+struct DoubleType
+{
+    explicit DoubleType(double d) : value(new double(d)) {}
+
+    ~DoubleType()
+    {
+        delete value;
+        value = nullptr;
+    }
+
+    DoubleType& add( double rhs );
+    DoubleType& subtract( double rhs );
+    DoubleType& multiply( double rhs );
+    DoubleType& divide(double rhs);
+    operator double() const {return *value;}
+    DoubleType& pow(const FloatType&);
+    DoubleType& pow(const DoubleType&);
+    DoubleType& pow(const IntType&); 
+    DoubleType& pow(double rhs);
+
+    private:
+        double* value = nullptr;
+        DoubleType& powInternal(const double exp);
+        
```

Comment:
align the `private` keyword with the left edge of the class's curly braces.
The private variables and functions should not be indented further than the public variables and functions
- [ ] revise